### ISSUE TITLE
Test failures in corefx_baseline leg on Unix do not cause the CI job to fail

### ISSUE THREAD

**Issue by jkotas** [dotnet/coreclr#16063]
There are several tests failing right now (e.g. https://github.com/dotnet/coreclr/issues/16062), yet the Jenkins UI shows it as green.

**Comment by jkotas** [comment]
cc @RussKeldorph @BruceForstall

**Comment by jkotas** [comment]
The failure is not detected because of this:
```
13:00:00 Command execution failed with exit code 1.
13:00:01 [Current build status] check if current [SUCCESS] is worse or equals then [SUCCESS] and better or equals then [SUCCESS]
13:00:01 Run condition [Current build status] enabling perform for step [[Archive the artifacts]]
13:00:01 Archiving artifacts
```

@dotnet/dnceng Could you please help me with where the broken logic that maps exit code 1 to success is? As far as I can tell, it is somewhere in Jenkins but I am not able to find it.

**Comment by mmitche** [comment]
@jkotas Do you have a link?

**Comment by jkotas** [comment]
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_ubuntu_corefx_baseline_prtest/178/consoleText

**Comment by mmitche** [comment]
@jkotas I'm pretty sure this is not jenkins.  Without exception, if the exit code from a build step is nonzero, the build fails.

The "Command execution failed with exit code 1" isn't printed by Jenkins, it's printed by the coreclr scripts, so it's entirely possible that this script printed that and then exited with 0.  What's actually a little interesting here is that I've seen something similar happen from time to time on the VSTS side.  The build will fail, print the same thing, then continue onward as if nothing happened.  I've never been able to repro locally but it happens often enough to make me think that something is up.

**Comment by BruceForstall** [comment]
Looks like the `tests\scripts\run-corefx-tests.py` script executes build-tests.sh:
```
[/mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/tests/scripts/run-corefx-tests.py]: ./build-tests.sh -Release -os:Linux -buildArch:x64 -- /p:WithoutCategories=IgnoreForCI 
```

And it should return whatever exit code build-tests.sh returns:
```
returncode = 0 if testing else os.system(command)
sys.exit(returncode)
```

(testing is False).

Jenkins looks at the return code from the python command:
```
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_ubuntu_corefx_baseline_prtest/configure
```

Presumably python is correctly setting the error code based on the script.

build-tests.sh is a deep test of sh + run.exe stuff...

**Comment by mmitche** [comment]
I didn't reproduce the original error but saw something very similar on an attempt.   crossgen crashed and returned 0 from python:

```
/mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prt                                                                          est$ python -u ./tests/scripts/run-corefx-tests.py -arch x64 -ci_arch x64 -build_type Checked -fx_root ./_/fx -fx_branch mast                                                                          er -env_script ./set_stress_test_env.sh
[./tests/scripts/run-corefx-tests.py]: Configuration:
[./tests/scripts/run-corefx-tests.py]:  arch: x64
[./tests/scripts/run-corefx-tests.py]:  ci_arch: x64
[./tests/scripts/run-corefx-tests.py]:  build_type: Checked
[./tests/scripts/run-corefx-tests.py]:  clr_root: /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_                                                                          baseline_prtest
[./tests/scripts/run-corefx-tests.py]:  fx_root: _/fx
[./tests/scripts/run-corefx-tests.py]:  fx_branch: master
[./tests/scripts/run-corefx-tests.py]:  fx_commit: HEAD
[./tests/scripts/run-corefx-tests.py]:  env_script: /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_coref                                                                          x_baseline_prtest/set_stress_test_env.sh
[./tests/scripts/run-corefx-tests.py]: git clone -b master --single-branch https://github.com/dotnet/corefx.git _/fx
Cloning into '_/fx'...
remote: Counting objects: 579608, done.
remote: Compressing objects: 100% (18/18), done.
remote: Total 579608 (delta 2), reused 2 (delta 2), pack-reused 579588
Receiving objects: 100% (579608/579608), 157.14 MiB | 23.20 MiB/s, done.
Resolving deltas: 100% (450415/450415), done.
[./tests/scripts/run-corefx-tests.py]: [cd] _/fx
[./tests/scripts/run-corefx-tests.py]: git checkout HEAD
Your branch is up-to-date with 'origin/master'.
[./tests/scripts/run-corefx-tests.py]: HOME=_/fx/tempHome
[./tests/scripts/run-corefx-tests.py]: ./build.sh -Release -os:Linux -buildArch:x64 -- /p:CoreCLROverridePath=/mnt/j/workspac                                                                          e/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/bin/Product/Linux.x64.Checked
Installing dotnet cli...
Restoring BuildTools version 2.1.0-preview2-02502-01...
Using RID linux-x64 for BuildTools native tools
Initializing BuildTools...
Making all .sh files executable under Tools.
Tools/crossgen.sh: line 21: NoWarn: command not found
  Restoring packages for /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/Tool                                                                          s/crossgen/crossgen.csproj...
  Installing Microsoft.NETCore.DotNetAppHost 2.0.4.
  Installing Microsoft.NETCore.DotNetHostResolver 2.0.4.
  Installing Microsoft.NETCore.App 2.0.4.
  Installing Microsoft.NETCore.DotNetHostPolicy 2.0.4.
  Installing NETStandard.Library 2.0.1.
  Installing runtime.linux-x64.Microsoft.NETCore.DotNetAppHost 2.0.4.
  Installing runtime.linux-x64.Microsoft.NETCore.DotNetHostResolver 2.0.4.
  Installing runtime.linux-x64.Microsoft.NETCore.DotNetHostPolicy 2.0.4.
  Installing runtime.linux-x64.Microsoft.NETCore.App 2.0.4.
/mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/Tools/crossgen/crossgen.cspro                                                                          j : warning NU1603: crossgen depends on Microsoft.NETCore.App (>= 2.0.3) but Microsoft.NETCore.App 2.0.3 was not found. An ap                                                                          proximate best match of Microsoft.NETCore.App 2.0.4 was resolved.
/mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/Tools/crossgen/crossgen.cspro                                                                          j : error NU1103: Unable to find a stable package Microsoft.NETCore.Platforms with version (>= 2.0.1)
/mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/Tools/crossgen/crossgen.cspro                                                                          j : error NU1103:   - Found 683 version(s) in https://dotnet.myget.org/F/dotnet-core/api/v3/index.json [ Nearest version: 2.1                                                                          .0-dev-di-25909-02 ]
  Generating MSBuild file /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/Too                                                                          ls/crossgen/obj/crossgen.csproj.nuget.g.props.
  Generating MSBuild file /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/Too                                                                          ls/crossgen/obj/crossgen.csproj.nuget.g.targets.
  Restore failed in 2.14 sec for /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_                                                                          /fx/Tools/crossgen/crossgen.csproj.
Done initializing tools.

Unhandled Exception: System.IO.FileLoadException: Could not load file or assembly 'System.Threading, Version=4.1.0.0, Culture                                                                          =neutral, PublicKeyToken=b03f5f7f11d50a3a'. The located assembly's manifest definition does not match the assembly reference.                                                                           (Exception from HRESULT: 0x80131040)
   at System.Console.get_Out()
   at System.ConsolePal.RefreshColors(ConsoleColor& toChange, ConsoleColor value)
   at Microsoft.DotNet.Execute.Setup.ExecuteCommand(String commandSelectedByUser, List`1 parametersSelectedByUser)
   at Microsoft.DotNet.Execute.Executor.Main(String[] args)
/mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/run.sh: line 17: 43330 Aborte                                                                          d                 (core dumped) $__dotnet $__toolRuntime/run.exe $__scriptpath/config.json "$@"
dotnet-bot@ubuntu1404-20170925-e23400:/mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prt                                                                          est$ echo $?
0

```

**Comment by BruceForstall** [comment]
I wonder if the Linux dotnet CLI tool isn't setting/propagating exit code correctly.

**Comment by jkotas** [comment]
I am adding extra temporary logging in https://github.com/dotnet/corefx/pull/26866 . I am not able to reproduce the problem locally.

**Comment by jkotas** [comment]
I have confirmed that the `dotnet run` command returns exit code correctly. So the error must be getting lost somewhere in the shell/python goo.

**Comment by jkotas** [comment]
This should be fixed by #16225
